File: garden/seeds/java/app/src/main/java/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/activity/BaseActivity.java
package com.{{company_name}}.android.{{app_package_name_prefix}}.activity;

import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.{{company_name}}.android.{{app_package_name_prefix}}.{{app_class_prefix}}App;

import butterknife.ButterKnife;

public abstract class BaseActivity extends RxActivity {

    protected {{app_class_prefix}}App app;

    /**
     * @return the layout resource to use for this activity,
     * or a value <= 0 if no layout should be used
     */
    @LayoutRes protected abstract int getLayoutResource();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        app = ({{app_class_prefix}}App) getApplicationContext();


        final int layoutResId = getLayoutResource();
        if (layoutResId > 0) {
            setContentView(layoutResId);
        }
        ButterKnife.bind(this);
    }

    /**
     * Retrieve a fragment from {@linkplain #getSupportFragmentManager()}
     *
     * @param id  The layout id of the fragment to find
     * @param cls The class of the fragment
     * @param <T>
     * @return A fragment found in the layout with the given id, or null
     */
    @SuppressWarnings("unchecked")
    @Nullable
    protected <T extends Fragment> T findFragment(@IdRes int id, Class<T> cls) {
        return (T) getSupportFragmentManager().findFragmentById(id);
    }
}

File: garden/seeds/kotlin/app/src/main/kotlin/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/activity/BaseActivity.kt
package com.{{company_name}}.android.{{app_package_name_prefix}}.activity

import android.content.Context
import android.os.Bundle
import android.support.annotation.LayoutRes
import android.view.MenuItem
import com.{{company_name}}.android.{{app_package_name_prefix}}.{{app_class_prefix}}AppImpl
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper

public abstract class BaseActivity : RxActivity() {

    private lateinit var app: {{app_class_prefix}}AppImpl

    /**
     * @return the layout resource to use for this activity,
     * * or a value <= 0 if no layout should be used
     */
    @LayoutRes protected abstract fun getLayoutResource(): Int

    override protected fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase))
    }

    override protected fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        app = getApplicationContext() as {{app_class_prefix}}AppImpl

        val layoutResId = getLayoutResource()
        if (layoutResId > 0) {
            setContentView(layoutResId)
        }
    }

    protected fun getApp() : {{app_class_prefix}}AppImpl {
        return app
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true;
        } else {
            return super.onOptionsItemSelected(item)
        }
    }
}

